docs/specs/sections/protocol.tex

```diff
@@ -0,0 +1,79 @@
+
+The workflow is depicted in Figure \ref{fig:protocol}, and described with full details as follows.
+
+\begin{figure}[h]
+	\centering
+		\includegraphics[width=390pt,draft=false]{images/protocol.eps}
+	\caption{Overview of the protocol messages exchanged between the user, the Dusk Network, and the SP.}
+	\label{fig:protocol}
+\end{figure}
+
+\begin{enumerate}
+	\item (\textbf{user}) $\mathsf{send\_license\_req}$ : compute a license stealth address $(\lpk, R)$ belonging to the user, using the user's own public key, and also an additional key $\ksym_{\License} = \hb(\lsk)G$, by computing first the user's $\lsk$. Then, compute the request stealth address $(\rpk, R)$ and $\kdh$ using the SP's public key. And finally send the following request to the network:
+
+
+		$$\Request = ((\rpk, R), \enc, \nonce)$$,
+
+	where 
+
+		$$\enc = \Enc_{\kdh} ((\lpk, R)||\ksym_{\License}; \nonce)$$.
+
+
+	\item (\textbf{SP}) $\mathsf{get\_license\_req}$ : continuously check the network for incoming license requests.
+
+	\item (\textbf{SP}) $\mathsf{send\_license}$ : upon receiving a request from a user, define a set of attributes $\attr$ representing the license, and compute a digital signature as follows:
+
+		$$\lsig = \sign_{\sk_{\SP}}(\lpk, \attr)$$.
+
+	Then, send the following license to the network:
+
+		$$\License = ((\lpk, R), \enc, \nonce, \pos)$$,
+
+	where 
+
+		$$\enc = \Enc_{\ksym_{\License}} (\lsig || \attr; \nonce)$$.
+
+	\item (\textbf{user}) $\mathsf{get\_license}$ : receive the license by scanning the incoming transactions.
+
+	\item (\textbf{user}) $\mathsf{call\_nullify\_license}$ : when desiring to use the license, nullify it by executing a call to the license contract. The following steps are performed:
```

Comment:
I think this should be a separate workflow.
One workflow is for getting a license and another one is to use it.

Reply:
We will create another section to depict the workflows at the code level. I'd stick with it for a general overview.